Update rules_pkg to the latest release.

aiuto opened the pull request:
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type

What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
pkg_tar is being referenced at a legacy location (@rules_pkg//:pkg.bzl) which will go away in a future release of rules_pkg. This PR updates rules_pkg to the latest release and fixes the references to be to the correct location (@rules_pkg//pkg:tar.bzl) which will remain correct indefinitely.

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

uhthomas commented:
Looks good, though we may be using an old version of rules_pkg by the looks of it.

```
(14:33:39) ERROR: error loading package 'testdata': Label '@rules_pkg//pkg:tar.bzl' is invalid because 'pkg' is not a package; perhaps you meant to put the colon here: '@rules_pkg//:pkg/tar.bzl'?
```

aiuto commented:
Fixed.
I updated to the latest rules_pkg, and fixed the tests which were depending on the ancient, broken, behavior of adding a '.' folder to every tarball.
